extensions/api/data-management/contractnegotiation/src/main/java/org/eclipse/dataspaceconnector/api/datamanagement/contractnegotiation/service/ContractNegotiationServiceImpl.java

```diff
@@ -0,0 +1,117 @@
+import org.eclipse.dataspaceconnector.spi.contract.negotiation.ConsumerContractNegotiationManager;
+import org.eclipse.dataspaceconnector.spi.contract.negotiation.response.NegotiationResult;
+import org.eclipse.dataspaceconnector.spi.contract.negotiation.store.ContractNegotiationStore;
+import org.eclipse.dataspaceconnector.spi.iam.ClaimToken;
+import org.eclipse.dataspaceconnector.spi.query.QuerySpec;
+import org.eclipse.dataspaceconnector.spi.transaction.TransactionContext;
+import org.eclipse.dataspaceconnector.spi.types.domain.contract.agreement.ContractAgreement;
+import org.eclipse.dataspaceconnector.spi.types.domain.contract.negotiation.ContractNegotiation;
+import org.eclipse.dataspaceconnector.spi.types.domain.contract.negotiation.ContractNegotiationStates;
+import org.eclipse.dataspaceconnector.spi.types.domain.contract.negotiation.ContractOfferRequest;
+import org.eclipse.dataspaceconnector.spi.types.domain.contract.offer.ContractDefinition;
+
+import java.util.Collection;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static java.lang.String.format;
+import static java.util.stream.Collectors.toList;
+
+public class ContractNegotiationServiceImpl implements ContractNegotiationService {
+
+    private final ContractNegotiationStore store;
+    private final ConsumerContractNegotiationManager manager;
+    private final TransactionContext transactionContext;
+
+    public ContractNegotiationServiceImpl(ContractNegotiationStore store, ConsumerContractNegotiationManager manager, TransactionContext transactionContext) {
+        this.store = store;
+        this.manager = manager;
+        this.transactionContext = transactionContext;
+    }
+
+    @Override
+    public ContractNegotiation findbyId(String contractNegotiationId) {
+        return store.find(contractNegotiationId);
+    }
+
+    @Override
+    public Collection<ContractNegotiation> query(QuerySpec query) {
+        return store.queryNegotiations(query).collect(toList());
+    }
+
+    @Override
+    public String getState(String negotiationId) {
+        return Optional.of(negotiationId)
+                .map(this::findbyId)
+                .map(ContractNegotiation::getState)
+                .map(ContractNegotiationStates::from)
+                .map(Enum::name)
+                .orElse(null);
+    }
+
+    @Override
+    public ContractAgreement getForNegotiation(String negotiationId) {
+        return store.findContractAgreement(negotiationId);
+    }
+
+    @Override
+    public ContractNegotiation initiateNegotiation(ContractOfferRequest request) {
+        var result = manager.initiate(request);
+        return result.getContent();
+    }
+
+    @Override
+    public ServiceResult<ContractNegotiation> cancel(String negotiationId) {
+        var result = new AtomicReference<ServiceResult<ContractNegotiation>>();
+
+        transactionContext.execute(() -> {
```

Comment:
You can now directly return the result of the `execute` method without using an `AtomicReference`. See #968 .

Reply:
done